src/test/java/wolox/training/repositories/BookRepositoryTest.java

```diff
@@ -34,4 +35,16 @@ public void testFindFirstByAuthorWhenSaveSuccessTheBook() {
 
         assertEquals("Jaime", author.get().getAuthor());
     }
+
+    @Test
+    public void testFindAllByPublisherIsAndGenreIsAndYearwhenReturnABook() {
+        bookRepository.save(book());
+        String publisher = "War Car 2";
+        String genre = "Action";
+        String year = "1990";
+
+        List<Book> booksFound = bookRepository.findAllByPublisherIsAndGenreIsAndYear(publisher, genre, year);
+
+        assertEquals(1, booksFound.size());
```

Comment:
you can add more validations, like the genre or publisher or another parameter to be sure that the information is ok